### ISSUE TITLE
[Content]:Use specific examples to illustrate when Stores should be used instead of signals.

### ISSUE THREAD

**Issue by huangsheng7271** [solidjs/solid-docs#968]
### 📚 Subject area/topic

/concepts/stores.mdx

### 📋 Page(s) affected (or suggested, for new content)

The document says that Stores are very useful when handling complex states, but I feel that signals can also handle it. The document only mentions the usage of Stores. I think specific examples should be used to illustrate in which scenarios using Stores is a better choice than signals.

### 📋 Description of content that is out-of-date or incorrect

The document says that Stores are very useful when handling complex states, but I feel that signals can also handle it. The document only mentions the usage of Stores. I think specific examples should be used to illustrate in which scenarios using Stores is a better choice than signals.

**Comment by brenelz** [comment]
I feel almost as soon as you have nested objects you'd want to use a store, but not sure what others mental model for using stores vs signals is

**Comment by atilafassina** [comment]
@brenelz 

> I feel almost as soon as you have nested objects you'd want to use a store, but not sure what others mental model for using stores vs signals is

Does it need to be nested? Any complex type would do (arrays or objects) nested or not.
If your state has more than 1 value, it's a store.

e.g.:

```ts
const [user, setUser] = createSignal("brenelz")
```

vs

```ts
const [user, setUser] = createStore( { id: "rec-123", handler: "brenelz" })
```


## suggested action item

- [ ] Add a note that Stores are recommended for any kind of complex type objects

**Comment by brenelz** [comment]
@atilafassina yeah stores work in that case too but if its just one flattened object I might still use a signal. Not sure if thats ideal tho

**Comment by atilafassina** [comment]
@brenelz it depends on how it's updated and where.

the problem of having an object or array in a signal is that every time you update a single value, the whole thing gets recreated, so we lose fine-grained reactivity.

maybe in a few use-cases that's ok, but I find that more often than not, I want my updates to be as granular as possible.

**Comment by brenelz** [comment]
@atilafassina yeah thats a fair point. I should probably use stores more often then
